features/levering-koopsom.feature

```diff
@@ -0,0 +1,132 @@
+                }
+            }
+        }
+        """
+
+
+    Scenario: Onroerendezaak (met koopsom & koopjaar & aardCultuurbebouwd) wordt opgevraagd met fields parameter met 1 van die drie velden (koopsom of koopjaar of aardCultuurbebouwd) erin
+        Als onroerendezaak gezocht word met /kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870487970000?fields=koopsom
+        Dan worden de koopsom & koopjaar & aardCultuurBebouwd teruggegeven
+        En wordt de volgende json response teruggegeven:
+        """
+        {
+            "identificatie": "NL.IMKAD.KadastraalObject.76870487970000",
+            "koopsom": {
+                "koopsom": 185000,
+                "koopjaar": 2015
+            },
+            "type": "perceel",
+            "aardCultuurBebouwd": {
+                "code": "11",
+                "waarde": "Wonen"
+            },
+            "_links": {
+                "self": {
+                    "href": "/kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870487970000?fields=koopsom"
+                }
+            }
+        }
+        """
+
+    Scenario: Onroerendezaak (met koopsom & koopjaar & aardCultuurbebouwd) wordt opgevraagd met fields parameter zonder 1 van die drie velden (koopsom of koopjaar of aardCultuurbebouwd) erin
+        Als onroerendezaak gezocht word met /kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870487970000?fields=kadastraleAanduiding
+        Dan worden de koopsom & koopjaar & aardCultuurBebouwd niet teruggegeven
+        En wordt de volgende json response teruggegeven:
+        """
+        {
+            "identificatie": "NL.IMKAD.KadastraalObject.76870487970000",
+            "type": "perceel",
+            "kadastraleAanduiding": "Beekbergen K 4879",
+            "_links": {
+                "self": {
+                    "href": "/kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870487970000?fields=kadastraleAanduiding"
+                }
+            }
+        }
+        """
+
+    Scenario: Onroerendezaak (met alleen koopsom & aardCultuurbebouwd) wordt opgevraagd zonder fields parameter
+        Als onroerendezaak gezocht word met /kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870488070000
+        Dan worden de koopsom & aardCultuurBebouwd niet teruggegeven
+        En wordt de volgende json response teruggegeven:
+        """
+        {
+            "identificatie": "NL.IMKAD.KadastraalObject.76870488070000",
+            "type": "perceel",
+            "kadastraleAanduiding": "Beekbergen K 4880",
+            "_links": {
+                "self": {
+                    "href": "/kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870488070000"
+                }
+            }
+        }
+        """
+
+    Scenario: Onroerendezaak (met alleen koopsom & aardCultuurbebouwd) wordt opgevraagd met fields parameter met 1 van die twee velden (koopsom of aardCultuurbebouwd) erin
```

Comment:
De vorige scenario, deze scenario en de volgende scenario kunnen worden samengevoegd tot één abstract scenario:

    Abstract Scenario: Onroerendezaak (met alleen koopsom & aardCultuurbebouwd) wordt opgevraagd
        Als onroerendezaak gezocht word met /kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870488070000<fields>
        Dan worden de koopsom & aardCultuurBebouwd niet teruggegeven
        En wordt de volgende json response teruggegeven:
        """
        {
            "identificatie": "NL.IMKAD.KadastraalObject.76870488070000",
            "type": "perceel",
            "kadastraleAanduiding": "Beekbergen K 4880",
            "_links": {
                "self": {
                    "href": "/kadastraalonroerendezaken/NL.IMKAD.KadastraalObject.76870488070000"
                }
            }
        }
        """

        Voorbeelden:
        | fields                       | omschrijving                                                                           |
        |                              | zonder fields parameter                                                                |
        | ?fields=koopsom              | met fields parameter met 1 van die twee velden (koopsom of aardCultuurbebouwd) erin    |
        | ?fields=kadastraleAanduiding | met fields parameter zonder 1 van die twee velden (koopsom of aardCultuurbebouwd) erin |

Reply:
Zoals het nu is gespecificeerd, is het een alles of niets (zie ook opmerking van Frank).

Verder zie ik dat ik heb te strak samengevoegd. Ik zie dat de scenario's

- Onroerendezaak (met koopsom & koopjaar & aardCultuurbebouwd) wordt opgevraagd met fields parameter zonder 1 van die drie velden (koopsom of koopjaar of aardCultuurbebouwd) erin
- Onroerendezaak (met alleen koopsom & aardCultuurbebouwd) wordt opgevraagd met fields parameter zonder 1 van die twee velden (koopsom of aardCultuurbebouwd) erin

dezelfde response hebben en dus met een abstract scenario kunnen worden samengevoegd en scenario **Onroerendezaak (met alleen koopsom & aardCultuurbebouwd) wordt opgevraagd met fields parameter met 1 van die twee velden (koopsom of aardCultuurbebouwd) erin** niet. Hier wordt met fields geen kadastrale aanduiding gevraagd en krijgt je hem ook niet.